.eslintrc.cjs

```diff
@@ -1,23 +1,55 @@
 module.exports = {
+  extends: [
+    'plugin:@typescript-eslint/recommended-type-checked',
+    'plugin:@typescript-eslint/stylistic-type-checked',
+    'prettier',
+  ],
   parser: '@typescript-eslint/parser',
-  extends: ['plugin:@typescript-eslint/recommended', 'prettier'],
+  parserOptions: {
+    project: ['./packages/*/tsconfig.json', './tsconfig.eslint.json'],
+    tsconfigRootDir: __dirname,
+  },
   plugins: ['@typescript-eslint', 'prettier', 'no-only-tests'],
   rules: {
+    // These off/configured-differently-by-default rules fit well for us
+    '@typescript-eslint/no-unused-vars': ['error', { ignoreRestSiblings: true }],
+    'no-only-tests/no-only-tests': 'error',
+    '@typescript-eslint/no-shadow': ['error'],
+    'no-console': 'warn',
+
+    /*
+    interface: 404 results in 198 files
+    type: 129 results in 73 files
+    */
+    '@typescript-eslint/consistent-type-definitions': 'off',
+
+    // Todo: do we want these?
+    '@typescript-eslint/await-thenable': 'off',
     '@typescript-eslint/ban-ts-comment': 'off',
-    '@typescript-eslint/camelcase': 'off',
-    '@typescript-eslint/explicit-module-boundary-types': 'off',
+    '@typescript-eslint/class-literal-property-style': 'off',
+    '@typescript-eslint/consistent-indexed-object-style': 'off',
+    '@typescript-eslint/dot-notation': 'off',
+    '@typescript-eslint/no-base-to-string': 'off',
+    '@typescript-eslint/no-confusing-void-expression': 'off',
     '@typescript-eslint/no-empty-function': 'off',
-    '@typescript-eslint/no-explicit-any': 'off',
-    '@typescript-eslint/no-non-null-assertion': 'off',
-    '@typescript-eslint/no-unused-vars': 'off',
-    '@typescript-eslint/no-use-before-define': 'off',
-    '@typescript-eslint/no-var-requires': 'off',
+    '@typescript-eslint/no-floating-promises': 'off',
+    '@typescript-eslint/no-misused-promises': 'off',
     '@typescript-eslint/no-this-alias': 'off',
-    'no-console': 'warn',
+    '@typescript-eslint/no-unsafe-argument': 'off',
+    '@typescript-eslint/no-unsafe-assignment': 'off',
+    '@typescript-eslint/no-unsafe-call': 'off',
+    '@typescript-eslint/no-unsafe-member-access': 'off',
+    '@typescript-eslint/no-unsafe-return': 'off',
+    '@typescript-eslint/prefer-nullish-coalescing': 'off',
+    '@typescript-eslint/prefer-string-starts-ends-with': 'off',
+    '@typescript-eslint/require-await': 'off',
+    '@typescript-eslint/restrict-plus-operands': 'off',
+    '@typescript-eslint/restrict-template-expressions': 'off',
+    '@typescript-eslint/sort-type-constituents': 'off',
+    '@typescript-eslint/unbound-method': 'off',
```

Comment:
Each of these had at least a few existing warnings, so I didn't want to overreach and solve for them without asking. Are any of these rules ones you feel particularly strongly about?

- https://typescript-eslint.io/await-thenable
- https://typescript-eslint.io/ban-ts-comment
- https://typescript-eslint.io/class-literal-property-style
- https://typescript-eslint.io/consistent-indexed-object-style
- https://typescript-eslint.io/dot-notation
- https://typescript-eslint.io/no-base-to-string
- https://typescript-eslint.io/no-confusing-void-expression
- https://typescript-eslint.io/no-empty-function
- https://typescript-eslint.io/no-floating-promises
- https://typescript-eslint.io/no-misused-promises
- https://typescript-eslint.io/no-this-alias
- https://typescript-eslint.io/no-unsafe-argument
- https://typescript-eslint.io/no-unsafe-assignment
- https://typescript-eslint.io/no-unsafe-call
- https://typescript-eslint.io/no-unsafe-member-access
- https://typescript-eslint.io/no-unsafe-return
- https://typescript-eslint.io/prefer-nullish-coalescing
- https://typescript-eslint.io/prefer-string-starts-ends-with
- https://typescript-eslint.io/require-await
- https://typescript-eslint.io/restrict-plus-operands
- https://typescript-eslint.io/restrict-template-expressions
- https://typescript-eslint.io/sort-type-constituents
- https://typescript-eslint.io/unbound-method

If not, I can go ahead and fix for all their complaints.

Reply:
> [typescript-eslint.io/await-thenable](https://typescript-eslint.io/await-thenable)

I'll keep it turned off. We have cases where we don't want to `await` a promise on purpose (middleware is one).

> [typescript-eslint.io/no-empty-function](https://typescript-eslint.io/no-empty-function)

Maybe we have a few no-ops in the code base? Not sure. I'd like to add this rule if possible and fix the cases where we trigger the error.

Reply:
> https://typescript-eslint.io/rules/ban-ts-comment

This will cause issues when we reference experimental packages without generated types ([see here](https://github.com/withastro/astro/blob/main/packages/integrations/markdoc/src/index.ts#L16-L17)). I think these comments are rare, but have use cases we talk about during PR reviews. I'd prefer keeping this turned off!

Reply:
Same comment as above, would prefer that we don't change the rules in an upgrade PR.

Reply:
> > [typescript-eslint.io/rules/ban-ts-comment](https://typescript-eslint.io/rules/ban-ts-comment)
> 
> This will cause issues when we reference experimental packages without generated types ([see here](https://github.com/withastro/astro/blob/main/packages/integrations/markdoc/src/index.ts?rgh-link-date=2023-06-20T11%3A05%3A25Z#L16-L17)). I think these comments are rare, but have use cases we talk about during PR reviews. I'd prefer keeping this turned off!

Maybe helpful?: it can be configured to allow the comment directives _if_ there's a description attached.

> Same comment as above, would prefer that we don't change the rules in an upgrade PR.

👍 I'll keep these all as _todo_.